### PR TITLE
Add http vertx client test cases to http/vertx scenario

### PR DESCRIPTION
**Pull request by jcarranzan**
### Summary

- Tweak vertx http client request in the test cases.
- Added http2 protocol and request with name parameter test cases.

Please select the relevant options.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Dependency update
- [ ] Refactoring
- [ ] Backport
- [x] New scenario (non-breaking change which adds functionality)
- [ ] This change requires a documentation update
- [ ] This change requires execution against OCP (use `run tests` phrase in comment)

### Checklist:
- [ ] Methods and classes used in PR scenarios are meaningful
- [x] Commits are well encapsulated and follow [the best practices](https://cbea.ms/git-commit/)

### CODE REVIEW

**Comment by QuarkusQE** [comment]
Can one of the admins verify this patch?

File: http/vertx/src/test/java/io/quarkus/ts/vertx/AbstractVertxIT.java [hunk 1]
```diff
@@ -21,13 +21,19 @@
...
 import io.vertx.core.http.HttpMethod;
+import io.vertx.core.http.HttpVersion;
+import io.vertx.core.json.JsonObject;
 
 public abstract class AbstractVertxIT {
 
     @QuarkusApplication
     static final RestService service = new RestService();
 
+    private String localhost = service.getHost().replace("http://", "");
```
Comment (michalvavrik) [inline http/vertx/src/test/java/io/quarkus/ts/vertx/AbstractVertxIT.java:35]: Wouldn't `io.quarkus.test.bootstrap.Protocol#NONE` do the trick? Sorry, didn't try it, but this feels like a hack.

Comment (jcarranzan) [reply http/vertx/src/test/java/io/quarkus/ts/vertx/AbstractVertxIT.java:35]: Well, I know what you feel about the "hack" but I've checked Protocol class and NONE option but that doesn't do the trick. If you know any quick way in quarkus test framework please tell me and I will change it. Thanks

Comment (michalvavrik) [reply http/vertx/src/test/java/io/quarkus/ts/vertx/AbstractVertxIT.java:35]: Nah, not that important. I'll try it when I find a little time, but your explanation is ok.

Comment (michalvavrik) [reply http/vertx/src/test/java/io/quarkus/ts/vertx/AbstractVertxIT.java:35]: @jcarranzan I take it back, `service.getURI(Protocol.NONE).getHost()` does the trick.

Comment (michalvavrik) [reply http/vertx/src/test/java/io/quarkus/ts/vertx/AbstractVertxIT.java:35]: please use it instead of what I call the hack :-)

Comment (jcarranzan) [reply http/vertx/src/test/java/io/quarkus/ts/vertx/AbstractVertxIT.java]: oh I see, it has been changed as you mentioned, thank you @michalvavrik !

**Comment by michalvavrik** [comment]
@jcarranzan just one more thing - these 3 commits don't feel right either, for in 8c8b2292f920d79f3f41dbec7bb6555127e3e262 you introduce change and in bc3e8aec8c0e9e26d71f44d17d59489ea2df0e47 change the previously introduced code, let's squash these 2 pls (+ there was a formatting error I think are going to address).

**Review by michalvavrik — APPROVED** [review]
Thanks for the contribution (and 3 commits :-))

**Comment by jcarranzan** [comment]
> Thanks for the contribution (and 3 commits :-))

I was squashing them right now but you were quicker doing the merge, thanks @michalvavrik

**Pull request merged**